### PR TITLE
bpo-36654: add example to generate token from another file

### PR DESCRIPTION
**Pull request by Windsooon**
Add example to generate token from another file.

https://bugs.python.org/issue36654

### CODE REVIEW

**Review by eamanu — COMMENTED** [review]
This example confuse me, because the file ```example.py``` does not  exist. Maybe you need to do something like ```hello.py```

File: Doc/library/tokenize.rst [hunk 1]
```diff
@@ -278,3 +278,15 @@ The exact token type names can be displayed using the :option:`-e` option:
     4,10-4,11:          RPAR           ')'
     4,11-4,12:          NEWLINE        '\n'
     5,0-5,0:            ENDMARKER      ''
+
+Example of tokenizing from another file::
```
Comment (berkerpeksag) [inline Doc/library/tokenize.rst]: Changing this sentence to something like "Example of tokenizing a file programmatically::" or "Example of using :func:\`tokenize.tokenize\` programmatically::" would be better.

See for a similar example:

https://github.com/python/cpython/blob/5c403b203510549a3f89d138d3265c5cc0cc12af/Lib/trace.py#L32-L48

File: Doc/library/tokenize.rst [hunk 2]
```diff
@@ -278,3 +278,15 @@ The exact token type names can be displayed using the :option:`-e` option:
...
+
+    import tokenize
+    f = open('example.py', 'rb')
```
Comment (berkerpeksag) [inline Doc/library/tokenize.rst]: Using `tokenize.open()` would be better as it automatically detect the encoding of the file:

https://github.com/python/cpython/blob/5d90954856cc1d999dd5d153b807055fde0d53ea/Lib/tokenize.py#L449

Comment (berkerpeksag) [reply Doc/library/tokenize.rst]: Also, I agree with Emmanuel that using the existing `hello.py` as an example would make the example easier to follow.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

File: Doc/library/tokenize.rst (hunk 2 above)
Comment (Windsooon) [reply Doc/library/tokenize.rst]: I found the [tokenize function](https://github.com/python/cpython/blob/master/Lib/tokenize.py#L404) already call `detect_encoding`.

Comment (berkerpeksag) [reply Doc/library/tokenize.rst]: I just realized that `tokenize.open()` returns a text stream, so passing it to `tokenize.tokenize()` wouldn't work.

We can use `tokenize.generate_tokens()` to demonstrate how the `str` API works:

```py
import tokenize

with tokenize.open('hello.py') as f:
    token_gen = tokenize.generate_tokens(f.readline)
    for token in token_gen:
        print(token)
```

Then fallback to your example to show the usage of the `bytes` API:

```py
import tokenize

with open('hello.py', 'rb') as f:
    token_gen = tokenize.tokenize(f.readline)
    for token in token_gen:
        print(token)
```

**Comment by matrixise** [comment]
Hi @Windsooon 

Could you update your PR with the last comment of @berkerpeksag and with the `with` statement?

Thank you

**Review by matrixise — DISMISSED** [review]
Thanks for your contribution but you have to update your PR with the recommendations of @berkerpeksag

**Comment by Windsooon** [comment]
Hello, @matrixise, I would love to update the example if needed. However, Serhiy Storchaka didn't agree with it

> I do not think a new example is needed. The existing example already demonstrates the use of file's readline method.

https://bugs.python.org/issue36654

**Comment by csabella** [comment]
@Windsooon, any updates?  Thanks!

**Comment by Windsooon** [comment]
Sure, I will fix it in two days. Thank you for reminding.

**Comment by Windsooon** [comment]
Hi, @csabella. I just updated the PR base on @berkerpeksag 's suggestion.

**Review by berkerpeksag — APPROVED** [review]
Looks pretty good to me, thank you!

**Pull request merged**